wire-library/wire-runtime-swift/src/main/swift/wellknowntypes/Duration.swift

```diff
@@ -119,10 +118,38 @@ extension Duration : Proto3Codable {
 
 #if !WIRE_REMOVE_CODABLE
 extension Duration : Codable {
-    public enum CodingKeys : String, CodingKey {
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: Duration.CodingKeys.self)
```

Comment:
@efirestone this is incorrect. Since this is generated code, I'm not sure how you want to proceed.

Timestamp | string | "1972-01-01T10:00:20.021Z" | Uses RFC 3339, where generated output will always be Z-normalized and uses 0, 3, 6 or 9 fractional digits. Offsets other than "Z" are also accepted.
-- | -- | -- | --
Duration | string | "1.000340012s", "1s" | Generated output always contains 0, 3, 6, or 9 fractional digits, depending on required precision, followed by the suffix "s". Accepted are any fractional digits (also none) as long as they fit into nano-seconds precision and the suffix "s" is required.

Reply:
Resolved in https://github.com/square/wire/pull/2380